cyclonedx-cocoapods.gemspec

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require_relative "lib/cyclonedx/cocoapods/version"
+
+Gem::Specification.new do |spec|
+  spec.name          = "cyclonedx-cocoapods"
+  spec.version       = CycloneDX::CocoaPods::VERSION
+  spec.authors       = ["José González"]
+  spec.email         = ["jose.gonzalez@openinput.com"]
+
+  spec.summary       = "CycloneDX software bill-of-material (SBoM) generation utility"
+  spec.description   = "CycloneDX is a lightweight software bill-of-material (SBOM) specification designed for use in application security contexts and supply chain component analysis. This Gem generates CycloneDX BOMs from CocoaPods projects."
+  spec.homepage      = "https://github.com/CycloneDX/cyclonedx-cocoapods"
+  spec.license       = "Apache-2.0"
+  spec.required_ruby_version = Gem::Requirement.new(">= 2.4.0")
+
+  spec.metadata["homepage_uri"] = spec.homepage
+  spec.metadata["source_code_uri"] = "https://github.com/CycloneDX/cyclonedx-cocoapods.git"
+
```

Comment:
I checked `cyclonedx-ruby` to find an example but I couldn't find any metadata in their `gemspec`. Do you mean adding something like the following?
```ruby
spec.metadata['component'] = "https://cocoapods.org/"
```
I don't understand how this value could be derived.

Reply:
The metadata is about the BOM and the top level component it describes.

Reply:
An example from .NET, which I haven’t immediately implemented yet, if generating a BOM for a specific project file you could determine some stuff about the top level component the BOM describes. 

Something that has been implemented is adding the .NET tool as one of the tools in the metadata.

Reply:
Ah, ok! You are talking about the [metadata node in the BOM](https://cyclonedx.org/docs/1.2/#type_metadata), aren't you? I got confused because the comment was made in the `metadata` of the `gemspec`.

Ok, let's see if I understood it correctly... You want me to populate the BOM's `metadata` including `cyclonedx-cocoapods` as a [tool](https://cyclonedx.org/docs/1.2/#type_toolType), am I right? And you are saying that maybe that information could be included in the `gemspec`'s `metadata`? 🤔 I'm no expert at creating gems, but I would say that you don't have access to the `gemspec` inside the gem's code.

Regarding the component that the BOM describes, I don't think that's feasible... A `Podfile` doesn't describe the project itself, instead it usually points to a Xcode workspace. That workspace may contain several projects and/or targets with several build configurations, and the Podfile (among other things) specifies the dependencies for each of those targets/build configurations. I don't see how to automatically extract information from a Xcode workspace without knowing how that workspace is structured.

Reply:
> Regarding the component that the BOM describes, I don't think that's feasible... 

Understood. I was not sure it was possible either. Thank you for clarifying.

Reply:
I have added the list of tools in the metadata section, and I also have added parameters to the CLI command to include the information of the component for which the BOM is being generated in the metadata section